engine/src/main/java/org/hibernate/search/spatial/impl/Point.java

```diff
@@ -175,7 +175,12 @@ public double getDistanceTo(Point other) {
 	 * @return the distance between the points
 	 * @see <a href="http://www.movable-type.co.uk/scripts/latlong.html">Distance haversine formula</a>
 	 */
-	public double getDistanceTo(final double latitude,final double longitude) {
+	public double getDistanceTo(final Double latitude,final Double longitude) {
+		if ( latitude == null || longitude == null ) {
```

Comment:
In light of the other suggestions, maybe you can keep the original signature?
I don't like this one, as the invoker of the method should be responsible to not invoke this with null values, rather than the method implementation being responsibl to work around for such a case.